iree/compiler/Codegen/SPIRV/ConvertToSPIRVPass.cpp

```diff
@@ -49,32 +50,46 @@ namespace {
 // Resource utilities
 //===----------------------------------------------------------------------===//
 
+/// Gets a unique name for `originalName` in `moduleOp` to avoid symbol
+/// collision.
+std::string getUniqueNameInSymbolTable(ModuleOp moduleOp,
+                                       const std::string &originalName) {
+  // First check whether the original name is not used.
+  if (!SymbolTable::lookupSymbolIn(moduleOp, originalName)) return originalName;
```

Comment:
SymbolTable insert will do this for you, and instead of doing this n^2 will let you do it o(n) (you'll make the SymbolTable and then pass it to the functions that insert the symbols)

Reply:
(you can create the op with whatever name you want then as the base and then do symbolTable.insert(op) and it'll rename it for you if needed)

Reply:
Yup. It's a bit tricky though due to the deferred nature of conversion framework. But should work for our case here. Changed.